src/components/AttachmentModal.js

```diff
@@ -44,28 +56,78 @@ const propTypes = {
 const defaultProps = {
     title: '',
     sourceURL: null,
+    file: {name: null},
     onConfirm: null,
-    isAuthTokenRequired: false,
+    currentAction: {
+        actionName: '', person: [], sequenceNumber: 0, timestamp: 0, message: []
+    },
+    children: null,
+    setAttachmentModalData: () => { },
+    getNextAttachment: () => { }
 };
 
 class AttachmentModal extends Component {
     constructor(props) {
         super(props);
-
         this.state = {
             isModalOpen: false,
             file: null,
             sourceURL: props.sourceURL,
         };
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.sourceURL !== this.props.sourceURL) {
+            this.setCurrentModalData(this.props);
+        }
+        if (this.state.isModalOpen) {
+            document.addEventListener('keydown', this.handleKeyPress, false);
+        } else {
+            document.removeEventListener('keydown', this.handleKeyPress, false);
+        }
+    }
+
+    setCurrentModalData(props) {
+        this.setState({
+            sourceURL: props.sourceURL,
+            file: props.file,
+            isAuthTokenRequired: props.currentAction.isAttachment,
+            isModalOpen: true
+        });
+    }
+
+    getAttachmentThumbnail(htmlString) {
+        const imgRegx = /<img.*?src="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    getAttachmentSource(htmlString) {
+        const imgRegx = /<img.*?data-expensify-source="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    handleKeyPress = (event) => {
+        if (event.key === 'ArrowRight' || event.key === 'ArrowLeft') {
+            event.stopPropagation();
+            const nextAttachment = this.props.getNextAttachment(this.props.currentAction, event.key === 'ArrowRight');
```

Comment:
Would be better if the method was named `getNextAttachmentAction` and actually had given us an action than referring to it via nextAttachment.action. It seems like the only value in this is nextAttachment.action and so it would be better if it was directly accessed in a variable like nextAttachmentAction.